storage/client/file/src/main/java/com/azure/storage/file/FileServiceClientBuilder.java

```diff
@@ -3,7 +3,309 @@
+     * @throws IllegalStateException If neither a {@link SharedKeyCredential} or {@link SASTokenCredential} has been set.
+     */
+    public FileServiceClient buildClient() {
+        return new FileServiceClient(buildAsyncClient());
+    }
+
+    /**
+     * Sets the endpoint for the Azure Storage File instance that the client will interact with.
+     *
+     * <p>Query parameters of the endpoint will be parsed using {@link SASTokenCredential#fromQuery(String) fromQuery} in an
+     * attempt to generate a {@link SASTokenCredential} to authenticate requests sent to the service.</p>
+     *
+     * @param endpoint The URL of the Azure Storage File instance to send service requests to and receive responses from.
+     * @return the updated FileServiceClientBuilder object
+     * @throws IllegalArgumentException If {@code endpoint} isn't a proper URL
+     */
+    public FileServiceClientBuilder endpoint(String endpoint) {
+        Objects.requireNonNull(endpoint);
+        try {
+            URL fullURL = new URL(endpoint);
+            this.endpoint = new URL(fullURL.getProtocol() + "://" + fullURL.getHost());
+
+            // Attempt to get the SAS token from the URL passed
+            SASTokenCredential credential = SASTokenCredential.fromQuery(fullURL.getQuery());
+            if (credential != null) {
+                this.sasTokenCredential = credential;
+            }
+        } catch (MalformedURLException ex) {
+            throw new IllegalArgumentException("The Azure Storage File Service endpoint url is malformed.");
+        }
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link SASTokenCredential} used to authenticate requests sent to the Queue service.
+     *
+     * @param credential SAS token credential generated from the Storage account that authorizes requests
+     * @return the updated FileServiceClientBuilder object
+     * @throws NullPointerException If {@code credential} is {@code null}.
+     */
+    public FileServiceClientBuilder credential(SASTokenCredential credential) {
+        this.sasTokenCredential = Objects.requireNonNull(credential);
+        return this;
+    }
+
+    /**
+     * Creates a {@link SharedKeyCredential} from the {@code connectionString} used to authenticate requests sent to the
+     * File service.
+     *
+     * @param connectionString Connection string from the Access Keys section in the Storage account
+     * @return the updated FileServiceClientBuilder object
+     * @throws NullPointerException If {@code connectionString} is {@code null}.
+     */
+    public FileServiceClientBuilder connectionString(String connectionString) {
+        Objects.requireNonNull(connectionString);
+        this.sharedKeyCredential = SharedKeyCredential.fromConnectionString(connectionString);
+        getEndPointFromConnectionString(connectionString);
+        return this;
+    }
+
+    private void getEndPointFromConnectionString(String connectionString) {
```

Comment:
Do you think we should put it into common? Will address this later.

Reply:
I'm not a fan of addressing things later. They will likely get forgotten. If you want to address later, please file an issue and reference it from your comment, so that we know it is being tracked. Even better would be to make the change as eagerly as possible, if you believe it is correct, such that it never needs to be logged or addressed in the future.

Reply:
I have this methods all across the queue and file, and I think blob needs to handle this as well.

This PR is super large. It is better to have it address them in another PR for all three service together instead of doing this one by one.


Here is the issue: https://github.com/Azure/azure-sdk-for-java/issues/4550